src/CirclesToken.sol

```diff
@@ -0,0 +1,124 @@
+  }
+
+  ///////
+  // Optional ERC20 Functions
+  // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+  ///////
+
+  // TODO: Optional - Figure out address to string conversion
+  //function name() view returns (string name) {
+  //  return "Circles" + person.toString();
+  //}
+
+  // TODO: Optional - Implement?
+  // IDK if there is something short enough that would make sense here
+  //function symbol view() returns (string symbol) {
+  //  return "CIR"
+  //}
+
+  // TODO: Optional - Choose, with issuanceRate
+  function decimals() view returns (uint8 decimals) {
+    return 18;
+  }
+
+  ///////
+  // Required ERC20 functions
+  // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+  ///////
+
+  //TODO: Choose, with decimals()
+  uint256 constant issuanceRate = 1;
+
+  //TODO: non-continuous payout?
+  function totalSupply() view returns (uint256 totalSupply) {
+    (now - rateUpdatedTimestamp) * issuanceRate
+  }
+
+  uint256 heldElsewhere;
+  mapping (address => uint256) balances;
+
+  function balanceOf(address _owner) view returns (uint256 balance) {
+    if (address == person) {
+      totalSupply() - heldElsewhere;
+    } else {
+      balances[tokenOwner];
+    }
+  }
+
+  event Transfer(address indexed _from, address indexed _to, uint256 _value)
+
+  function _transfer(address _from, address _to, _value) internal returns (bool success) {
+    require( balanceOf(_from) >= _value, "Insufficient balance" );
+
+    // decrement _from balance
+    if (_from == person) {
+      heldElsewhere = heldElsewhere + _value;
+    } else {
+      balances[_from] = balances[_from] - _value;
+    }
+
+    // increment _to balance
+    if (_to == person) {
+      heldElsewhere = heldElsewhere - _value;
+    } else {
+      balances[_to] = balances[_to] + _value;
+    }
+
+    Transfer(_from, _to, _value);
+    return true;
+  }
+
+  function transfer(address _to, uint256 _value) returns (bool success) {
+    return _transfer(msg.sender, _to, _value);
+  }
+
+  mapping (address => mapping (address = uint256)) public allowances;
+
+  function transferFrom(address _from, address _to, uint256 _value) returns (bool success) {
+    require( allowances[_from][msg.sender] >= _value, "Not authorized" );
+    allowances[_from][msg.sender] = allowances[_from][msg.sender] - _value;
+    return _transfer(msg.sender, _to, _value);
+  }
+
+  event Approval(address indexed _owner, address indexed _spender, uint256 _value)
+
+  function approve(address _spender, uint256 _value) returns (bool success) {
+    allowances[msg.sender][_spender] = _value;
+
+    Approval(msg.sender, _spender, _value);
+    return true;
+  }
+
+  function allowance(address _owner, address _spender) view returns (uint256 remaining) {
+    return allowances[_owner][_spender];
+  }
+
+  ///////
+  // Support trusted currency exchanges for transitive payments
+  ///////
+
+  // TODO: Worry about accounts that may have a __circles_approveExchange function?
```

Comment:
Oops, accidentally added a commit that marked Davids comment as "outdated". Here is David's comment:

> feels a bit strange that we have to trust the person in `CirclesPerson` and then also approve the transfer on the token itself.
> 
> Is there some way we could encode the trust graph in the allowances mapping?

Reply:
In what way are you saying we trust the person in CirclesPerson? I don't see any strange behavior that could be caused by re-entrancy. Also, remember, here the person is the person who is the owner of the tokens. You should be able to trust your own contract in this way.

The problem comes in if someone accidentally has a function who's abi's signature is the same as __circles_approveExchange that returns true when the owner really wasn't intending to interact with Circles at all. In this case the balance can be "stolen". I don't think this is really a problem because I perceive the likelyhood of a random contract holding circles to have a function named __circles_approveExchange (or having a function signature hash collision with the name) is pretty small.

The only potential problems I see are: What if solidity does something weird when this function isn't implemented (since circlesToken can be sent to any address, including end user accounts)? From my reading, this isn't a problem because this is just an address.call that will return false on failure: https://solidity.readthedocs.io/en/v0.4.24/units-and-global-variables.html#address-related, meaning the funds aren't moved.